lib/tapioca/compilers/dsl/config.rb

```diff
@@ -0,0 +1,104 @@
+# typed: strict
+# frozen_string_literal: true
+
+begin
+  require "config"
+rescue LoadError
+  return
+end
+
+module Tapioca
+  module Compilers
+    module Dsl
+      # `Tapioca::Compilers::Dsl::Config` generates RBI files for classes generated by the
+      # [`config`](https://github.com/rubyconfig/config) gem.
+      #
+      # The gem creates a `Config::Options` instance based on the settings files and/or
+      # env variables. It then assigns this instance to a constant with a configurable name,
+      # by default `Settings`. Application code uses methods on this constant to read off
+      # config values.
+      #
+      # For a setting file like the following:
+      # ```yaml
+      # ---
+      # github:
+      #   token: 12345
+      #   client_id: 54321
+      #   client_secret: super_secret
+      # ```
+      # and a `Config` setup like:
+      # ```ruby
+      # Config.setup do |config|
+      #   config.const_name = "AppSettings"
+      # end
+      # ```
+      # this generator will produce the following RBI file:
+      # ```rbi
+      # AppSettings = T.let(T.unsafe(nil), AppSettingsConfigOptions)
+      #
+      # class AppSettingsConfigOptions < ::Config::Options
+      #   sig { returns(T.untyped) }
+      #   def github; end
```

Comment:
How does the gem handle nested values (e.g.: `token`)? Could we return an object that has the nested keys as methods instead of `T.untyped`?

Reply:
I am returning `T.untyped` for exactly this reason. The way the library works is based on `OpenStruct` etc and is super dynamic, so it will be very messy to define a different type for each nested object to represent it properly.

We are not very interested in the typing here, hopefully people are not using these config values in very complex ways. We are more interested in the method calls being known.